src/hooks/useNavigation.ts

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+
+import { useHistory } from 'react-router';
+
+import * as paths from 'routes/paths';
+
+/**
+ * Access parameterized app navigation.
+ *
+ * @returns functions that create onClick handlers.
+ *
+ * Example:
+ * const { getToMap } = useNavigation();
+ * ...
+ * return (<Button onClick={getToMap({hightlight: '0x5453424324bbbecdea'})}>To map</Button>);
+ *
+ * The handler will respond to crtl & meta click for new window.
+ */
+export const useNavigation = () => {
```

Comment:
interesting! when do you think this should be used instead of an `<a>` tag?

Reply:
A single source of truth for navigation (sorta along with paths.ts)

You can use it like `getToProfile({ address: 'me' })()` if wanted.

when you use an `a` tag that changes style.. I could do a global clear style on a tags.

There is also the Link component, but then that is importing both something from paths and something from router..

I think that it would make sense to standardize something with router `state` that allows adding things that don't live in the url but maybe not because I think that highlighting already makes sense to be a search param